### PR TITLE
WIP - React Components Show React Logo

### PR DESCRIPTION
**Pull request by James300**
Associated Issue: #4151

### Summary of Changes

* Add React icon to React components tabs

### Screenshots
react-icon-tab-2
react-icon-tab

The icon seems a little small, thoughts?

### CODE REVIEW

File: src/components/Editor/Tabs.js [hunk 1]
```diff
@@ -482,7 +487,8 @@ export default connect(
       searchTabs: getSearchTabs(state),
       sourceTabs: getSourcesForTabs(state),
       activeSearch: getActiveSearch(state),
-      searchOn: getActiveSearch(state) === "source"
+      searchOn: getActiveSearch(state) === "source",
+      sourceMetaData: getSourceMetaData(state, 0)
```
Comment (James300) [inline src/components/Editor/Tabs.js]: @jasonLaster I was actually having a problem with how to fetch the `sourceId` for `getSourceMetaData`

File: src/components/Editor/Tabs.js [hunk 2]
```diff
@@ -482,7 +489,8 @@ export default connect(
       searchTabs: getSearchTabs(state),
       sourceTabs: getSourcesForTabs(state),
       activeSearch: getActiveSearch(state),
-      searchOn: getActiveSearch(state) === "source"
+      searchOn: getActiveSearch(state) === "source",
+      sourceMetaData: { isReactComponent: true }
```
Comment (James300) [inline src/components/Editor/Tabs.js]: Hard-coded for now so I could complete the rest of the task, what should go here because I can't figure it out? 😢

Comment (bomsy) [reply src/components/Editor/Tabs.js]: @James300 You can do this

```diff
@@ -484,13 +484,15 @@ class SourceTabs extends PureComponent<Props, State> {

 export default connect(
   state => {
+    const selectedSource = getSelectedSource(state);
+    const sourceId = selectedSource ? selectedSource.get("id") : "";
     return {
-      selectedSource: getSelectedSource(state),
+      selectedSource,
       searchTabs: getSearchTabs(state),
       sourceTabs: getSourcesForTabs(state),
       activeSearch: getActiveSearch(state),
       searchOn: getActiveSearch(state) === "source",
-      sourceMetaData: { isReactComponent: true }
+      sourceMetaData: getSourceMetaData(state, sourceId)
     };
   },
   dispatch => bindActionCreators(actions, dispatch)
```
I think there is still the work to set the sourceMetaData in redux

Comment (James300) [reply src/components/Editor/Tabs.js]: Thank Bomsy this did it 👍

**Comment by James300** [comment]
I noticed a few problems with the framework `isReactComponent`:
- React.Component does not get recognized as a component
- Using require instead of import does not work

Also, the React Icon does not show when reloading previously opened files.

File: src/components/Editor/Tabs.css [hunk 3]
```diff
@@ -102,6 +102,15 @@
   background: var(--theme-highlight-blue);
 }
 
+.source-tab img.react {
+  mask: url(/images/react.svg) no-repeat;
+  mask-size: 100%;
+  padding-top: 12px;
+  height: 14px;
+  width: 14px;
+  background: var(--theme-highlight-blue);
```
Comment (jasonLaster) [inline src/components/Editor/Tabs.css]: dark mode might benefit from a different color... something darker...

Comment (James300) [reply src/components/Editor/Tabs.css]: image
image

Comment (James300) [reply src/components/Editor/Tabs.css]: This is using bluegrey, thoughts?

Comment (jasonLaster) [reply src/components/Editor/Tabs.css]: i like it

**Pull request merged**